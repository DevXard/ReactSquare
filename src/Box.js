const Box = ({ id, color, width, height, removeSquare}) => {
    
    const handleRemove = (id) => {
        removeSquare(id)
    }

    return (
        <div style={{ backgroundColor:`${color}`, height:`${height}px`, width:`${width}px`}}>
            <button onClick={ () => handleRemove(id)}>X</button>
        </div>
    )
}

export default Box;

