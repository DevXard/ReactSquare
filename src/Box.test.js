import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

it("Should render without crashing", function () {
  render(<Box />);
});

it("Should match snapshot", function () {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

