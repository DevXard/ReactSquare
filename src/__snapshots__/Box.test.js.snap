// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`Should match snapshot: 
    <DocumentFragment>
      <div>
        <button>
          X
        </button>
      </div>
    </DocumentFragment>
   1`] = `
<DocumentFragment>
  <div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
